zkevm-circuits/src/evm_circuit/execution/end_block.rs

```diff
@@ -0,0 +1,38 @@
+use crate::evm_circuit::{
+    execution::ExecutionGadget,
+    step::ExecutionState,
+    util::constraint_builder::ConstraintBuilder,
+    witness::{Block, Call, ExecStep, Transaction},
+};
+use eth_types::Field;
+use halo2_proofs::{circuit::Region, plonk::Error};
+use std::marker::PhantomData;
+
+#[derive(Clone, Debug)]
+pub(crate) struct EndBlockGadget<F> {
+    _marker: PhantomData<F>,
+}
+
+impl<F: Field> ExecutionGadget<F> for EndBlockGadget<F> {
```

Comment:
Is this implementation complete or just a place holder?  From the specs I see checks for `total_tx` and `total_rw`.  If this is a placeholder could you add a comment with `TODO`?